src/main/java/org/datadog/jenkins/plugins/datadog/DatadogGlobalConfiguration.java

```diff
@@ -174,67 +155,41 @@ public class DatadogGlobalConfiguration extends GlobalConfiguration {
     @DataBoundConstructor
     public DatadogGlobalConfiguration() {
         load(); // Load the persisted global configuration
-        loadEnvVariables(); // Load environment variables after as they should take precedence.
+        loadEnvVariables(); // Load environment variables
+    }
+
+    @Override
+    protected XmlFile getConfigFile() {
+        return new XmlFile(XSTREAM, new File(Jenkins.get().getRootDir(), getId() + ".xml"));
```

Comment:
I think we should have a mechanism to avoid breaking the Jenkins config for previous versions:
- User updates to the new Jenkins plugin version with the config refactor.
- User modifies config and saves the new config. The file format changes.
- User tries to downgrade Jenkins plugin to previous version. Cannot start as the file format is unexpected.

A possible approach could be:
- Store config with a new name (`globalConfig_v2.xml`) or something like that
- Try to load v2 config first. If it's not found, try to load legacy config.

With that approach, old Jenkins plugin version should still working.

However, not sure about the Config-as-Code format 🤔

Reply:
That's a great idea! I implemented the `_v2` storing/loading.
As for CasC, the config reads from it but does not write, meaning YAMLs will remain the same unless users change them manually (the new code is compatible with the old YAMLs). So it's not that big a problem as the one with XMLs.